elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java

```diff
@@ -116,16 +118,46 @@ public EntityBinding(EntityDictionary dictionary, Class<?> cls, String type, Str
 
         // Map id's, attributes, and relationships
         List<AccessibleObject> fieldOrMethodList = new ArrayList<>();
-        fieldOrMethodList.addAll(Arrays.asList(cls.getFields())
+        fieldOrMethodList.addAll(Arrays.asList(cls.getDeclaredFields())
                 .stream()
                 .filter((field) -> ! Modifier.isStatic(field.getModifiers()))
+                .filter((field) -> ! ((Field) field).isSynthetic())
                 .collect(Collectors.toList()));
 
-        fieldOrMethodList.addAll(Arrays.asList(cls.getMethods())
+        if (fieldOrMethodList.stream().anyMatch(field -> field.isAnnotationPresent(Id.class))) {
+            accessType = AccessType.FIELD;
+
+
+            /* Add all public methods that are computed */
+            fieldOrMethodList.addAll(Arrays.asList(cls.getMethods())
+                    .stream()
+                    .filter((method) -> !Modifier.isStatic(method.getModifiers()))
+                    .filter((method) -> method.isAnnotationPresent(ComputedAttribute.class)
+                            || method.isAnnotationPresent(ComputedRelationship.class))
+                    .collect(Collectors.toList()));
+
+            //Elide needs to manipulate private fields that are exposed.
+            fieldOrMethodList.forEach(field -> field.setAccessible(true));
+
+        } else {
+            /* Preserve the behavior of Elide 4.2.6 and earlier */
+            accessType = AccessType.PROPERTY;
+
+            fieldOrMethodList.clear();
+
+            /* Add all public fields */
+            fieldOrMethodList.addAll(Arrays.asList(cls.getFields())
                 .stream()
-                .filter((method) -> ! Modifier.isStatic(method.getModifiers()))
+                .filter((field) -> ! Modifier.isStatic(field.getModifiers()))
                 .collect(Collectors.toList()));
 
+            /* Add all public methods */
+            fieldOrMethodList.addAll(Arrays.asList(cls.getMethods())
+                    .stream()
+                    .filter((method) -> !Modifier.isStatic(method.getModifiers()))
+                    .collect(Collectors.toList()));
```

Comment:
I think we should extract part of this out into a function, since it appears 4 times in this diff.

```java
private <T extends AccessibleObject> List<T> getInstanceMembers(T[] objects, Predicate<T> filteredBy) {
    return Arrays.stream(objects)
        .filter(o -> !Modifier.isStatic(o.getModifiers()))
        .filter(filteredBy)
        .collect(Collectors.toList());
}
```

Reply:
o.getModifiers won't work here because it is not a common inherited method from AccessibleObject (It belongs in both Field and Method but not in AccessibleObject).

Reply:
I think I found another way though...